opensearch/src/main/java/org/opensearch/sql/opensearch/storage/script/filter/FilterQueryBuilder.java

```diff
@@ -96,16 +100,71 @@ public QueryBuilder visitFunction(FunctionExpression func, Object context) {
         return buildBoolQuery(func, context, BoolQueryBuilder::should);
       case "not":
         return buildBoolQuery(func, context, BoolQueryBuilder::mustNot);
+      case "nested":
+        // TODO Fill in case when adding support for syntax - nested(path, condition)
+        throw new SyntaxCheckException(
+            "Invalid syntax used for nested function in WHERE clause: "
+                + "nested(field | field, path) OPERATOR LITERAL"
+        );
       default: {
         LuceneQuery query = luceneQueries.get(name);
         if (query != null && query.canSupport(func)) {
           return query.build(func);
+        } else if (query != null && query.isNestedFunction(func)) {
+          return handleNestedPredicateExpression(func, query);
         }
         return buildScriptQuery(func);
       }
     }
   }
 
+  /**
+   * Build query for 'nested' function used in predicate expression. Supports 'nested' function on
+   * left and literal on right.
+   * @param func Function expression.
+   * @param query Function with nested function as first argument.
+   * @return Nested query.
+   */
+  private QueryBuilder handleNestedPredicateExpression(FunctionExpression func, LuceneQuery query) {
+    FunctionExpression nestedFunc = (FunctionExpression) func.getArguments().get(0);
+
+    // Do semantic checks to ensure valid parameters.
+    if (!(nestedFunc.getArguments().size() <= 2)) {
+      throw new IllegalArgumentException(
+          "nested function supports 2 parameters (field, path) or 1 parameter (field)"
+      );
+    }
+    for (int i = 0; i < nestedFunc.getArguments().size(); i++) {
+      if (!(nestedFunc.getArguments().get(i) instanceof ReferenceExpression)) {
+        throw new IllegalArgumentException(
+            String.format("Illegal nested field name: %s",
+                nestedFunc.getArguments().get(i).toString()
+            )
+        );
+      }
+    }
+
+    QueryBuilder innerQuery = query.buildNested(func);
+    boolean hasPathParam = nestedFunc.getArguments().size() == 2;
+    String pathStr = hasPathParam ? nestedFunc.getArguments().get(1).toString() :
+        getNestedPathString((ReferenceExpression) nestedFunc.getArguments().get(0));
+    return QueryBuilders.nestedQuery(pathStr, innerQuery, ScoreMode.None);
+  }
+
+  /**
+   * Dynamically generate path for nested field. An example field of 'office.section.cubicle'
+   * would dynamically generate the path 'office.section'.
+   * @param field nested field to generate path for.
+   * @return path for nested field.
+   */
+  private String getNestedPathString(ReferenceExpression field) {
+    String ret = "";
+    for (int i = 0; i < field.getPaths().size() - 1; i++) {
+      ret += (i == 0) ? field.getPaths().get(i) : "." + field.getPaths().get(i);
+    }
+    return ret;
```

Comment:
```suggestion
return String.join(".", field.getPaths());
```

I believe that does the same thing.

Reply:
We want to omit the last index to get the fields `path` value.